dag_in_context/src/interpreter.rs

```diff
@@ -193,6 +193,14 @@ impl<'a> VirtualMachine<'a> {
                 self.memory.insert(pointer.addr(), val);
                 Value::StateV
             }
+            TernaryOp::Select => {
+                let get_bool = |e: &RcExpr, vm: &mut Self| vm.interp_bool_expr(e, arg);
+                if get_bool(e1, self) {
+                    self.interpret_expr(e2, arg)
```

Comment:
We should think carefully here: do we need to interpret both, then select between the two?
I guess not, since all the effects will still happen due to the state edge? This is probably correct, might be worth a comment